The NETRC environment variable is honoured

alloveras opened the pull request:
The current documentation wording seems to suggest that the only `.netrc` file that will be considered by Terraform is the one sitting in the current user's HOME directory. However, unless I am missing something, Terraform uses `go-getter` to fetch remote modules which mean that the `NETRC` environment variable will also be respected and, in fact, will take precedence over any `.netrc` file on the user's home directory.

See: https://github.com/hashicorp/go-getter/blob/f7a8c48a1f863bcd167b51adcec00a18d10c85a5/netrc.go#L23-L36

Fixes https://github.com/hashicorp/terraform/issues/30136

## Draft CHANGELOG entry

### ENHANCEMENTS

-  The documentation now specifies that, when set, the environment variable `NETRC` will be honored.

hashicorp-cla commented:
[![CLA assistant check](https://cla.hashicorp.com/pull/badge/not_signed)](https://cla.hashicorp.com/hashicorp/terraform?pullRequest=32007)


Thank you for your submission! We require that all contributors sign our Contributor License Agreement ("CLA") before we can accept the contribution. [Read and sign the agreement](https://cla.hashicorp.com/hashicorp/terraform?pullRequest=32007)

[Learn more about why HashiCorp requires a CLA and what the CLA includes](https://www.hashicorp.com/cla)

<sub>Have you signed the CLA already but the status is still pending? [Recheck it](https://cla.hashicorp.com/check/hashicorp/terraform?pullRequest=32007).</sub>

alloveras commented on website/docs/language/modules/sources.mdx:
```diff
@@ -356,7 +356,9 @@ In either case, the result is interpreted as another module source address
 
 If an HTTP/HTTPS URL requires authentication credentials, use a `.netrc`
-file in your home directory to configure these. For information on this format,
+to configure these. By default, Terraform will search for the `.netrc` file 
```
```suggestion
file to configure these. By default, Terraform will search for the `.netrc` file 
```

alloveras commented on website/docs/language/modules/sources.mdx:
```diff
@@ -356,7 +356,9 @@ In either case, the result is interpreted as another module source address
-file in your home directory to configure these. For information on this format,
+to configure these. By default, Terraform will search for the `.netrc` file 
+in your HOME directory. However, that default filesystem location can be overriden 
+through the `NETRC` environment variable. For information on the `.netrc` format,
```
```suggestion
by setting the `NETRC` environment variable. For information on the `.netrc` format,
```

crw commented:
Thanks for the submission! I've notified the docs team for review.

mgarrell777 commented on website/docs/language/modules/sources.mdx:
```diff
@@ -356,7 +356,9 @@ In either case, the result is interpreted as another module source address
 
 If an HTTP/HTTPS URL requires authentication credentials, use a `.netrc`
-file in your home directory to configure these. For information on this format,
+file to configure these. By default, Terraform will search for the `.netrc` file 
+in your HOME directory. However, that default filesystem location can be overriden 
+by setting the `NETRC` environment variable. For information on the `.netrc` format,
 see [the documentation for using it in `curl`](https://everything.curl.dev/usingcurl/netrc).
```
```suggestion
file to configure the credentials. By default, Terraform searches for the `.netrc` file 
in your HOME directory. However, you can override the default filesystem location by setting the `NETRC` environment variable. For information on the `.netrc` format,
refer to [the documentation for using it in `curl`](https://everything.curl.dev/usingcurl/netrc).
```

Suggest changing to "the credentials" for clarity and to avoid "These" without a subject. Suggest changing "will search" to "searches" to avoid a future tense style issue. Suggest changing to "you can override" to avoid a passive voice style issue. Suggest changing "see" to "refer to" to avoid ableist language.

mgarrell777 reviewed (approved):
@alloveras Thanks very much for this PR. I have a few small wording change suggestions. But overall it looks good to me.

alloveras commented:
Hey @mgarrell777,
thanks for the feedback. I've incorporated your suggestions 👍

alloveras commented:
Hey @crw @mgarrell777,
is there anything pending on my end to get this PR over the line? Unless I've missed something, I think I've addressed all the feedback 😄 

Thanks!

crw commented:
Thanks for the bump, I'll follow up with @mgarrell777!

pandom commented:
Can I please merge this? This is a documentation change that passes requests.
